astropy/table/table.py

```diff
@@ -1922,7 +1922,9 @@ def replace_column(self, name, col):
         if self[name].info.indices:
             raise ValueError('cannot replace a table index column')
 
-        t = self.__class__([col], names=[name])
+        # We deliberately use ``Table`` here since it should be sufficient
+        # and it is just meant as a temporary container for the column
+        t = Table([col], names=[name])
```

Comment:
I guess the goal here is to ensure the input can be turned into a column. A fairly safe replacement would seem to be
```
if not isinstance(getattr(col, 'info', None), BaseColumnInfo):
    col = Column(col, name=name)
cols = OrderedDict(self.columns)
cols[name] = col
self._init_from_cols(cols.values())
```

Reply:
Here's a test case (using current master) that should be included.  I don't think the above would pass this:
```
In [11]: t = simple_table()
In [12]: b = np.ma.MaskedArray([4, 5, 6])
In [13]: b[1] = np.ma.masked
In [14]: t.replace_column('b', b)
INFO: Upgrading Table to masked Table. Use Table.filled() to convert to unmasked table. [astropy.table.table]

In [15]: t
Out[15]: 
<Table masked=True length=3>
  a     b    c  
int64 int64 str1
----- ----- ----
    1     4    c
    2    --    d
    3     6    e
```
I've been thinking about a better solution but nothing great has come to mind.  I believe that the implementation currently in this PR will work, I'm just a little unhappy about the case where a `Quantity` gets converted to `Column` and then back to `Quantity` because of `Table`.  I thought about using `QTable` there instead, but that is worse because it can end up sending `Quantity` objects into `Table._init_from_cols`, which is not expecting that and will leave them as `Quantity`.

So maybe a `TimeSeries` method overload is best, where you use `QTable`.  Or better yet fix `TimeSeries` so it works with this existing method.  :smile:

Reply:
Maybe a combination of the two - use the `Table` only for objects that don't have `BaseColumnInfo`? I.e., something like
```
if not isinstance(getattr(col, 'info', None), BaseColumnInfo):
    col = Table([col], names=[name])[name]
...
```